packages/rocketchat-lib/server/lib/notifyUsersOnMessage.js

```diff
@@ -104,9 +104,24 @@ RocketChat.callbacks.add('afterSaveMessage', function(message, room) {
 	// Update all the room activity tracker fields
 	RocketChat.models.Rooms.incMsgCountAndSetLastMessageById(message.rid, 1, message.ts, RocketChat.settings.get('Store_Last_Message') && message);
 
-	// Update all other subscriptions to alert their owners but witout incrementing
+	// Update all other subscriptions to alert their owners but without incrementing
 	// the unread counter, as it is only for mentions and direct messages
 	RocketChat.models.Subscriptions.setAlertForRoomIdExcludingUserId(message.rid, message.u._id);
+	const subscriptions = RocketChat.models.Subscriptions.findByRoomIdForAlert(message.rid);
+	if (subscriptions) {
+		let userSubscribed = {};
+		let showHiddenRoom;
+		subscriptions.forEach(function(subscription) {
+			if (subscription.u._id !== message.u._id) {
+				userSubscribed = RocketChat.models.Users.findById(message.u._id);
+				showHiddenRoom = userSubscribed && userSubscribed.settings && userSubscribed.settings.preferences && userSubscribed.settings.preferences.showHiddenRoomsWhenUnreadMessages;
+				RocketChat.models.Subscriptions.setAlertForUserSubscribed(subscription.u._id, showHiddenRoom);
```

Comment:
The idea here is to set everyone to alert at line `109` and then set again here to alert true/false depending on each user preference?

Reply:
What happens when `showHiddenRoom` is undefined?

Reply:
The idea is to set everyone to alert, but open room only for one want this behavior. Fixed.